classes/lib.php

```diff
@@ -59,4 +65,74 @@ public static function validate_ip_against_config() {
             send_unknown($msg);
         }
     }
+
+    /**
+     * Records that the cache was pinged. This is useful for cache debugging.
+     * @param int $previousincache
+     * @param int $previousindb
+     * @param int $new
+     * @param string $where cron or web
+     */
+    public static function record_cache_pinged(int $previousincache, int $previousindb, int $new, string $where) {
+        try {
+            $details = array_merge(
+                self::get_cache_log_details(),
+                [
+                    'previousvalueindb' => $previousindb,
+                    'previousvalueincache' => $previousincache,
+                    'newvalue' => $new,
+                    'where' => $where,
+                ]
+            );
+
+            $event = cache_ping::create([
+                'other' => $details,
+                'contextid' => context_system::instance()->id,
+            ]);
+            $event->trigger();
+        } catch (Throwable $e) {
+            debugging($e, DEBUG_DEVELOPER);
+        }
+    }
+
+    /**
+     * Records that the cache was checked. This is used for debugging cache mismatches
+     * @param int $valueincache
+     * @param int $valueindb
+     * @param string $where web or cron
+     */
+    public static function record_cache_checked(int $valueincache, int $valueindb, string $where) {
+        try {
+            $details = array_merge(
+                self::get_cache_log_details(),
+                [
+                    'valueindb' => $valueindb,
+                    'valueincache' => $valueincache,
+                    'where' => $where,
+                ]
+            );
+
+            $event = cache_check::create([
```

Comment:
is there a reason this is an event and not something simpler like just a direct error_log() call? I'm thinking of less moving parts in the whole thing, eg with logstore_standardqueued not complicating things

Reply:
After discusssion - have changed to just error_log